### ISSUE TITLE
DISTINCT operator broken in AQL

### ISSUE THREAD

**Issue by nr23730** [ehrbase/ehrbase#50]
When I execute the following query
```
{
  "q": "SELECT DISTINCT e/ehr_id/value FROM EHR e WHERE e/ehr_status/subject/external_ref/id/value='myID'"
}
```
it results in an error due to a `org.springframework.jdbc.BadSqlGrammarException`:
```
{
  "error": "Could not process query, reason:org.springframework.jdbc.BadSqlGrammarException: Access database using Jooq; bad SQL grammar [select distinct on (\"ehr_id/value\") \"alias_91990184\".\"/ehr_id/value\" from (select \"ehr_join\".\"id\" as \"/ehr_id/value\" from \"ehr\".\"entry\" right outer join \"ehr\".\"composition\" as \"composition_join\" on \"composition_join\".\"id\" = \"ehr\".\"entry\".\"composition_id\" right outer join \"ehr\".\"ehr\" as \"ehr_join\" on \"ehr_join\".\"id\" = \"composition_join\".\"ehr_id\" join \"ehr\".\"status\" as \"status_join\" on \"status_join\".\"ehr_id\" = \"ehr_join\".\"id\" join \"ehr\".\"party_identified\" as \"subject_ref\" on \"subject_ref\".\"id\" = \"status_join\".\"party\" where (ehr.js_ehr_status(\"status_join\".\"ehr_id\")::json #>>'{subject,external_ref,id,value}'='myID')) as \"alias_91990184\"]; nested exception is org.postgresql.util.PSQLException: ERROR: column \"ehr_id/value\" does not exist\n  Hint: Perhaps you meant to reference the column \"alias_91990184./ehr_id/value\".\n  Position: 21",
  "status": "Bad Request"
}
```

Running the query without the `DISTINCT` operator works just fine and results in
```
{
  "q": "SELECT e/ehr_id/value FROM EHR e WHERE e/ehr_status/subject/external_ref/id/value='myID'",
  "name": null,
  "columns": [
    {
      "path": "/ehr_id/value",
      "name": "#0"
    }
  ],
  "rows": [
    [
      "e1d8645f-8842-4ae6-99eb-3f8a7c408fce"
    ]
  ]
}
```

The last time I used the operator (before 15/11/2019, probably 2956a97) it worked fine. Versions I tried, where it's broken are 20aa369 and 929e0e7.

**Comment by Geisterfalle** [comment]
Thanks for the pointer, it's great to receive your helpful feedback! We are currently working on integration tests for AQL for the CI pipeline and will surely add tests for DISTINCT as well. Hope we can fix this asap.

**Comment by nr23730** [comment]
It looked a bit more into this. It came with b71dfeb.

The executed query is
```
select distinct on ("ehr_id/value") "alias_91990184"."/ehr_id/value"
FROM (
	select "ehr_join"."id" as "/ehr_id/value" 
	from "ehr"."entry"
	right outer join "ehr"."composition" as "composition_join" on "composition_join"."id" = "ehr"."entry"."composition_id"
	right outer join "ehr"."ehr" as "ehr_join" on "ehr_join"."id" = "composition_join"."ehr_id"
	join "ehr"."status" as "status_join" on "status_join"."ehr_id" = "ehr_join"."id"
	join "ehr"."party_identified" as "subject_ref" on "subject_ref"."id" = "status_join"."party"
	where (ehr.js_ehr_status("status_join"."ehr_id")::json #>>'{subject,external_ref,id,value}'='myID')
) as "alias_91990184"
```

And it should be:
```
select distinct on ("/ehr_id/value") "alias_91990184"."/ehr_id/value"
FROM (
	select "ehr_join"."id" as "/ehr_id/value" 
	from "ehr"."entry"
	right outer join "ehr"."composition" as "composition_join" on "composition_join"."id" = "ehr"."entry"."composition_id"
	right outer join "ehr"."ehr" as "ehr_join" on "ehr_join"."id" = "composition_join"."ehr_id"
	join "ehr"."status" as "status_join" on "status_join"."ehr_id" = "ehr_join"."id"
	join "ehr"."party_identified" as "subject_ref" on "subject_ref"."id" = "status_join"."party"
	where (ehr.js_ehr_status("status_join"."ehr_id")::json #>>'{subject,external_ref,id,value}'='myID')
) as "alias_91990184"
```

So it's just the `/` in the `DISTINCT ON` statement that is missing... Hope that helps...

**Comment by Geisterfalle** [comment]
Resolved by #56
